### PR TITLE
MINOR: add unit test for Utils.murmur2

### PR DESCRIPTION
**Pull request by natemurthy**
I recently needed to port the `Utils.murmur` hash function to Go for a project I'm working on. I realized there were no definite test cases for the Java implementation that I could use to tell me whether my ported implementation was correct and returned the same results as the code in this project. And although there is test coverage of this hash function by way of a caller in the `testKeyPartitionIsStable` unit in `DefaultPartitionerTest`, the granularity of the test there is not sufficient for any developer to write a reliable port in another language of choice.

Fortunately, I found someone who ran into the same trouble as me:

https://stackoverflow.com/questions/48582589/porting-kafkas-murmur2-implementation-to-go

who provided test cases with their solution. I took the test cases from the above and codified them as unit test cases for this hash function in `UtilsTest`. In this way, any developer writing a custom Kafka client or server in any language can look at these reference cases to confirm the correctness of their implementations with absolute confidence that keys hashed with this murmur function (modulo some number of partitions) land on the right partition.

An open question to the maintainers (and to anyone with expertise in non-cryptographic hashing): do the cases I've given add enough surface area to cover the complete inner workings of this function's mappings?

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java
Comment (mimaison) [inline clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: Should we import it explicitly?

**Review by mimaison — APPROVED** [review]
I agree, the `murmur2()` behavior should be properly tested so 3rd party clients can easily re-implement it. Thanks for adding this test.

I've left a minor comment but LGTM

File: clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java
Comment (natemurthy) [reply clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: I can do that... Is this the reason the `clients:checkstyleTest` fails? I can't view the file to the report listing the exact violation.

Comment (natemurthy) [reply clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]:
```
13:36:48 FAILURE: Build failed with an exception.
13:36:48 
13:36:48 * What went wrong:
13:36:48 Execution failed for task ':clients:checkstyleTest'.
13:36:48 > Checkstyle rule violations were found. See the report at: file:///home/jenkins/jenkins-slave/workspace/kafka-pr-jdk8-scala2.11/clients/build/reports/checkstyle/test.html
13:36:48   Checkstyle files with violations: 1
13:36:48   Checkstyle violations by severity: [error:5]
```

Comment (mimaison) [reply clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: There are a few checkstyle failures:
```
[ant:checkstyle] [ERROR] .../UtilsTest.java:62:20: ',' is not followed by whitespace. [WhitespaceAfter]
[ant:checkstyle] [ERROR] .../UtilsTest.java:68:34: ',' is not followed by whitespace. [WhitespaceAfter]
[ant:checkstyle] [ERROR] .../UtilsTest.java:68:38: ',' is not followed by whitespace. [WhitespaceAfter]
[ant:checkstyle] [ERROR] .../UtilsTest.java:71:31: 'typecast' is not followed by whitespace. [WhitespaceAfter]
[ant:checkstyle] [ERROR] .../UtilsTest.java:71:61: 'typecast' is not followed by whitespace. [WhitespaceAfter]
```

You can get this output by running:
```
gradle 
./gradlew clients:test
```

Comment (natemurthy) [reply clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java]: Oh yes, seeing that now just pushed a style fix: deb060e.

As for the explicit import, I'm ok with what is here. We either have import declaration on line 35 or explicit import on line 62.

**Comment by natemurthy** [comment]
Not sure how this regression got introduced https://builds.apache.org/job/kafka-pr-jdk8-scala2.11/17834/testReport/junit/kafka.api/PlaintextConsumerTest/testLowMaxFetchSizeForRequestAndPartition/. Any ideas?

**Comment by mimaison** [comment]
retest this please

**Comment by natemurthy** [comment]
@mimaison ok to merge this in?

**Comment by mimaison** [comment]
@natemurthy I'm not a committer so I cannot merge it.

@omkreddy @rajinisivaram Can you take a look?

**Comment by mimaison** [comment]
Ping @omkreddy @rajinisivaram

**Review by omkreddy — COMMENTED** [review]
@natemurthy Thanks for the PR. LGTM.

**Pull request merged**
